### ISSUE TITLE
Cannot install or update jedi-vim

### ISSUE THREAD

**Issue by spapanik** [junegunn/vim-plug#500]
When it tries to install jedi-vim it calls `git clone --depth 1 --no-single-branch --progress --recursive https://github.com/davidhalter/jedi-vim.git -b master jedi-vim`, which fails with Cloning into 'jedi-vim'...

remote: Counting objects: 97, done.
remote: Compressing objects: 100% (59/59), done.
remote: Total 97 (delta 22), reused 65 (delta 11), pack-reused 0
Unpacking objects: 100% (97/97), done.
Checking connectivity... done.
Submodule 'jedi' (https://github.com/davidhalter/jedi.git) registered for path 'jedi'
Cloning into 'xxxxxxxxxxxxxxx/jedi-vim/jedi'...
error: no such remote ref 995a6531225ba0b65e1ff863d97e5404d989047b
Fetched in submodule path 'jedi', but it did not contain 995a6531225ba0b65e1ff863d97e5404d989047b. Direct fetching of that commit failed.

**Comment by junegunn** [comment]
Sounds like an issue of jedi-vim repository. Have you tried contacting the maintainer of the project?

**Comment by junegunn** [comment]
The command works on my machine.

```sh
> git clone --depth 1 --no-single-branch --progress --recursive https://github.com/davidhalter/jedi-vim.git -b master jedi-vim
Cloning into 'jedi-vim'...
remote: Counting objects: 97, done.
remote: Compressing objects: 100% (59/59), done.
remote: Total 97 (delta 22), reused 65 (delta 11), pack-reused 0
Unpacking objects: 100% (97/97), done.
Checking connectivity... done.
Submodule 'jedi' (https://github.com/davidhalter/jedi.git) registered for path 'jedi'
Cloning into 'jedi'...
remote: Counting objects: 26016, done.
remote: Compressing objects: 100% (193/193), done.
remote: Total 26016 (delta 115), reused 0 (delta 0), pack-reused 25823
Receiving objects: 100% (26016/26016), 7.98 MiB | 1.41 MiB/s, done.
Resolving deltas: 100% (18501/18501), done.
Checking connectivity... done.
Submodule path 'jedi': checked out '995a6531225ba0b65e1ff863d97e5404d989047b'
```

**Comment by spapanik** [comment]
```$ git --version
git version 2.9.0

**Comment by junegunn** [comment]
Hmm, interesting, I had no issue with git 2.6.3, but I see the same problem with git 2.9.0. I wonder if this new behavior is going to be fixed/improved in future releases. If it's not, I'll have to find a workaround :(

For now, shallow cloning can be turned off with `let g:plug_shallow = 0`. The config is not per-plugin but global, so there's currently no way to disable it just for jedi-vim.

**Comment by junegunn** [comment]
http://lists-archives.com/git/870111-git-2-9-0-clone-recursive-fails-on-cloning-a-submodule.html

**Comment by blueyed** [comment]
Works for me with git 2.10.2, so probably fixed in Git itself?
If jedi-vim could do something like mentioned in https://github.com/junegunn/vim-plug/issues/515#issuecomment-246888293, I'd be glad to help in that regard.

**Comment by spapanik** [comment]
It is fixed in git 2.10

**Comment by junegunn** [comment]
Thanks for the confirmation, I'll close this then.
